src/server.h

```diff
@@ -767,26 +769,31 @@ typedef struct clientReplyBlock {
     char buf[];
 } clientReplyBlock;
 
+/* State for the Slot to Key API, for a single slot. The keys in the same slot
+ * are linked together using dictEntry metadata. See also "Slot to Key API" in
+ * cluster.c. */
+struct clusterSlotToKeys {
+    uint64_t count;             /* Number of keys in the slot. */
+    dictEntry *head;            /* The first key-value entry in the slot. */
+};
+typedef struct clusterSlotToKeys clusterSlotsToKeysData[CLUSTER_SLOTS];
+
 /* Redis database representation. There are multiple databases identified
  * by integers from 0 (the default database) up to the max configured
  * database. The database number is the 'id' field in the structure. */
 typedef struct redisDb {
-    dict *dict;                 /* The keyspace for this DB */
-    dict *expires;              /* Timeout of keys with a timeout set */
-    dict *blocking_keys;        /* Keys with clients waiting for data (BLPOP)*/
-    dict *ready_keys;           /* Blocked keys that received a PUSH */
-    dict *watched_keys;         /* WATCHED keys for MULTI/EXEC CAS */
-    int id;                     /* Database ID */
-    long long avg_ttl;          /* Average TTL, just for stats */
-    unsigned long expires_cursor; /* Cursor of the active expire cycle. */
-    list *defrag_later;         /* List of key names to attempt to defrag one by one, gradually. */
+    dict *dict;                    /* The keyspace for this DB */
+    dict *expires;                 /* Timeout of keys with a timeout set */
+    dict *blocking_keys;           /* Keys with clients waiting for data (BLPOP)*/
+    dict *ready_keys;              /* Blocked keys that received a PUSH */
+    dict *watched_keys;            /* WATCHED keys for MULTI/EXEC CAS */
+    int id;                        /* Database ID */
+    long long avg_ttl;             /* Average TTL, just for stats */
+    unsigned long expires_cursor;  /* Cursor of the active expire cycle. */
+    list *defrag_later;            /* List of key names to attempt to defrag one by one, gradually. */
+    clusterSlotsToKeysData *slots_to_keys; /* Array of slots to keys. Only used in cluster mode (db 0). */
```

Comment:
i think the indentation changes you added are unwanted, it causes the diff to look as if you changed all these lines, when in fact you just added one (harder to review and also trace back history of things in git log).
considering that anyway, one line has a unique indentation to it, i think we can revert the others.

maybe the middle ground is to move the `expires_cursor` line to be one before the last, and then the first bulk of lines will have the old (short) indentation, and the last two will have the same long one?

or, i'm also ok with leaving all the existing lines as is.
i.e. the short ones have extra indentation, and the two long ones have their own way...